sdk/identity/identity/src/credentials/applicationCredential.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential } from "@azure/core-auth";
+
+import { TokenCredentialOptions } from "../client/identityClient";
+
+import { ChainedTokenCredential } from "./chainedTokenCredential";
+
+import { EnvironmentCredential } from "./environmentCredential";
+import { CredentialPersistenceOptions } from "./credentialPersistenceOptions";
+import { DefaultManagedIdentityCredential } from "./defaultAzureCredential";
+
+/**
+ * Provides options to configure the {@link ApplicationCredential} class.
+ */
+export interface ApplicationCredentialOptions
+  extends TokenCredentialOptions,
+    CredentialPersistenceOptions {
```

Comment:
It doesn't look like the `EnvironmentCredential` or `ManagedIdentityCredential` uses the persistence options (if I understand correctly...they don't end up using the msal node flow), so I think we could get away with just extending `TokenCredentialOptions`.

@willmtemple  Am I correct that those credentials don't use persistence?

Reply:
EnvironmentCredential passes the options to credentials that support persistence. ManagedIdentityCredential does not currently support persistence, but it will eventually.